pandas/io/formats/style.py

```diff
@@ -2423,6 +2440,50 @@ def set_table_styles(
             self.table_styles = table_styles
         return self
 
+    def set_descriptors(
+        self, descriptors: list[Descriptor | tuple[str, Descriptor]] | None = None
+    ) -> Styler:
+        """
+        Add header-level calculations to the output which describes the data.
+
+        .. versionadded:: 1.5.0
+
+        Parameters
+        ----------
+        descriptors : list of str, callables or 2-tuples of str and callable
+            If a string is given must be a valid Series method, e.g. "mean" invokes
+            Series.mean().
+
+            If a callable is given must accept a Series and return a scalar.
+
+            If a 2-tuple, must be a string used as the name of the row and a
+            callable or string as above.
+
+        Returns
+        -------
+        self : Styler
+
+        Examples
+        --------
+
+        >>> df = DataFrame([[1, 2], [3, 4]], columns=["A", "B"])
+        >>> def udf_func(s):
+        ...     return s.mean()
+        >>> styler = df.style.set_descriptors([
+        ...     "mean",
+        ...     Series.mean,
```

Comment:
The screen show shows no lable on this line? Was it not possible to have `.__name__` as a default test?